osu.Game/Screens/Menu/MainMenu.cs

```diff
@@ -244,10 +247,16 @@ public override void OnResuming(IScreen last)
 
         public override bool OnExiting(IScreen next)
         {
-            if (!exitConfirmed && dialogOverlay != null && !(dialogOverlay.CurrentDialog is ConfirmExitDialog))
+            if (!exitConfirmed && dialogOverlay != null)
             {
-                dialogOverlay.Push(new ConfirmExitDialog(confirmAndExit, () => exitConfirmOverlay.Abort()));
-                return true;
+                if (!(dialogOverlay.CurrentDialog is ConfirmExitDialog))
```

Comment:
i think the following will read better:

```suggestion
                if (dialogOverlay.CurrentDialog is ConfirmExitDialog exitDialog)
                {
                	exitConfirmed = true;
                	exitDialog.Buttons.First().Click();
                }
                else
                {
                	dialogOverlay.Push(new ConfirmExitDialog(confirmAndExit, () => exitConfirmOverlay.Abort()));
                    return true;
                }
```